### PR TITLE
site: add zhihu & yuque column card

### LINKED ISSUE #43087: 官网博客增加专栏链接
1. 文章增加专栏链接配置

```diff
  ---
  title: 静态方法之痛
  date: 2023-04-26
  author: zombieJ
+ column_url: https://zhuanlan.zhihu.com/p/633333904
  ---
```

2. 在 https://ant.design/docs/blog/why-not-static-cn 底部放一个『去专栏参与讨论』的链接，链接到上面。

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

- close #43087

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | site: add zhihu card |
| 🇨🇳 Chinese |          - |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 291d0fc</samp>

This pull request adds a feature to show a Zhihu card at the end of blog posts that are cross-posted to Zhihu columns. The feature involves a new component `ZhihuCard`, a theme slot `Content`, and a metadata field `column_url` in the blog post markdown files.

### <samp>🤖 Generated by Copilot at 291d0fc</samp>

*  Add a `ZhihuCard` component that renders a card with a link to a Zhihu column where the blog post is also published ([link](https://github.com/ant-design/ant-design/pull/43122/files?diff=unified&w=0#diff-bb56fb7403ea3eff11f6a85deccccc35cfda3adf54348a792f798862ab9ac75bR1-R83))
* Import and conditionally render the `ZhihuCard` component in the `Content` component based on the `column_url` field in the blog post metadata ([link](https://github.com/ant-design/ant-design/pull/43122/files?diff=unified&w=0#diff-1533ad458dddb5cacbf24062cc5d262ca684a57e2754bf4777f24f37411d5f82R19), [link](https://github.com/ant-design/ant-design/pull/43122/files?diff=unified&w=0#diff-1533ad458dddb5cacbf24062cc5d262ca684a57e2754bf4777f24f37411d5f82R276))
* Add the `column_url` field to the metadata of the blog post `why-not-static.zh-CN.md` with the URL of the Zhihu column ([link](https://github.com/ant-design/ant-design/pull/43122/files?diff=unified&w=0#diff-c80567f8dbd7b0deabc0aae11e928b3792d35905fc64d6e7197b7ec26105618eR5))

### CODE REVIEW

File: .dumi/theme/slots/Content/ZhihuCard.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
+import { css } from '@emotion/react';
+import { Button, Card } from 'antd';
+import React from 'react';
+import useLocale from '../../../hooks/useLocale';
+import useSiteToken from '../../../hooks/useSiteToken';
+
+const ANTD_IMG_URL =
+  'https://picx.zhimg.com/v2-3b2bca09c2771e7a82a81562e806be4d_200x0.jpg?source=d16d100b';
```
Comment (afc163) [inline .dumi/theme/slots/Content/ZhihuCard.tsx:8]: 图片有点糊

File: .dumi/theme/slots/Content/ZhihuCard.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      margin-top: 8px;
+    `,
+  };
+};
+
+const locales = {
+  cn: {
+    bigTitle: '文章被收录于专栏：',
+    subTitle: '一个 UI 设计体系',
+    buttonText: '参与讨论',
```
Comment (afc163) [inline .dumi/theme/slots/Content/ZhihuCard.tsx:64]:
```suggestion
    buttonText: '我有观点，前往专栏参与讨论',
```

File: .dumi/theme/slots/Content/ZhihuCard.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    card: css`
+      width: 100%;
+      margin: 40px 0;
+      transition: all 0.2s;
+      background-color: ${token.siteMarkdownCodeBg};
+      &:hover {
+        border-color: transparent;
+        box-shadow: 0 1px 2px -2px rgba(0, 0, 0, 0.16), 0 3px 6px 0 rgba(0, 0, 0, 0.12),
+          0 5px 12px 4px rgba(0, 0, 0, 0.1);
+      }
```
Comment (afc163) [inline .dumi/theme/slots/Content/ZhihuCard.tsx:22]: 这个 hover 效果感觉比较奇怪，建议去掉。

File: .dumi/theme/slots/Content/ZhihuCard.tsx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  'https://picx.zhimg.com/v2-3b2bca09c2771e7a82a81562e806be4d_200x0.jpg?source=d16d100b';
+
+const useStyle = () => {
+  const { token } = useSiteToken();
+  return {
+    card: css`
+      width: 100%;
+      margin: 40px 0;
+      transition: all 0.2s;
+      background-color: ${token.siteMarkdownCodeBg};
```
Comment (afc163) [inline .dumi/theme/slots/Content/ZhihuCard.tsx:17]: 这个颜色比截图里的深了点。

File: .dumi/theme/slots/Content/ZhihuCard.tsx (hunk 2 above)
Comment (li-jia-nan) [reply .dumi/theme/slots/Content/ZhihuCard.tsx:64]: 这个会不会按钮太长了？

File: .dumi/theme/slots/Content/ZhihuCard.tsx (hunk 1 above)
Comment (li-jia-nan) [reply .dumi/theme/slots/Content/ZhihuCard.tsx:8]: 大佬有没有其它图片？内容更丰富，更清晰一点的~

File: .dumi/theme/slots/Content/ZhihuCard.tsx (hunk 3 above)
Comment (li-jia-nan) [reply .dumi/theme/slots/Content/ZhihuCard.tsx:22]: OK

File: .dumi/theme/slots/Content/index.tsx [hunk 5]
```diff
@@ -272,6 +273,7 @@ const Content: React.FC<{ children: ReactNode }> = ({ children }) => {
           ) : null}
           {!meta.frontmatter.__autoDescription && meta.frontmatter.description}
           {children}
+          {meta?.frontmatter?.column_url && <ZhihuCard link={meta?.frontmatter?.column_url} />}
```
Comment (linxianxi) [inline .dumi/theme/slots/Content/index.tsx:276]: && 后面不用加可选链了，前面的我看上面也没加可选链？

File: .dumi/theme/slots/Content/ZhihuCard.tsx (hunk 2 above)
Comment (afc163) [reply .dumi/theme/slots/Content/ZhihuCard.tsx]: 好像还行。

File: .dumi/theme/slots/Content/ZhihuCard.tsx (hunk 1 above)
Comment (afc163) [reply .dumi/theme/slots/Content/ZhihuCard.tsx:8]: https://picx.zhimg.com/v2-3b2bca09c2771e7a82a81562e806be4d.jpg?source=d16d100b

File: .dumi/theme/slots/Content/ZhihuCard.tsx [hunk 6]
```diff
@@ -0,0 +1,100 @@
+import { css } from '@emotion/react';
+import { Button, Card } from 'antd';
+import React from 'react';
+import useLocale from '../../../hooks/useLocale';
+import useSiteToken from '../../../hooks/useSiteToken';
+
+const ANTD_IMG_URL =
+  'https://picx.zhimg.com/v2-3b2bca09c2771e7a82a81562e806be4d_200x0.jpg?source=d16d100b';
```
Comment (afc163) [inline .dumi/theme/slots/Content/ZhihuCard.tsx:8]:
```suggestion
  'https://picx.zhimg.com/v2-3b2bca09c2771e7a82a81562e806be4d.jpg?source=d16d100b';
```

File: .dumi/theme/slots/Content/index.tsx (hunk 5 above)
Comment (li-jia-nan) [reply .dumi/theme/slots/Content/index.tsx]: 改了

**Comment by li-jia-nan** [comment]
最终的效果：

image

image

File: .dumi/theme/slots/Content/ZhihuCard.tsx [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+  const { token } = useSiteToken();
+  return {
+    card: css`
+      width: 100%;
+      margin: 40px 0;
+      transition: all 0.2s;
+      background-color: ${token.colorFillQuaternary};
+      &:hover {
+        border-color: rgba(0, 0, 0, 0.1);
+      }
```
Comment (afc163) [inline .dumi/theme/slots/Content/ZhihuCard.tsx:20]: 这个边框也不好看，去掉试试？

File: .dumi/theme/slots/Content/ZhihuCard.tsx [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+const ZhihuCard: React.FC<Props> = ({ link }) => {
+  const [locale] = useLocale(locales);
+  const { card, bigTitle, cardBody, left, title, subTitle } = useStyle();
+  if (!link) {
+    return null;
+  }
+  return (
+    <Card css={card}>
```
Comment (afc163) [inline .dumi/theme/slots/Content/ZhihuCard.tsx:82]: `bordered={false}` 吧

File: .dumi/theme/slots/Content/ZhihuCard.tsx (hunk 7 above)
Comment (li-jia-nan) [reply .dumi/theme/slots/Content/ZhihuCard.tsx]: done~

File: .dumi/theme/slots/Content/ZhihuCard.tsx (hunk 8 above)
Comment (li-jia-nan) [reply .dumi/theme/slots/Content/ZhihuCard.tsx]: done~

**Comment by afc163** [comment]
也加个语雀的吧，卡片里加个 Divider 分隔开就行：https://www.yuque.com/ant-design/ant-design

File: .dumi/theme/slots/Content/ZhihuCard.tsx [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+      font-size: 14px;
+      font-weight: 400;
+      margin-top: 8px;
+    `,
+  };
+};
+
+const locales = {
+  cn: {
+    bigTitle: '文章被收录于知乎专栏：',
```
Comment (afc163) [inline .dumi/theme/slots/Content/ZhihuCard.tsx:57]:
```suggestion
    bigTitle: '文章被以下专栏收录：',
```

File: docs/blog/why-not-static.en-US.md [hunk 10]
```diff
@@ -2,6 +2,8 @@
 title: Pain of static methods
 date: 2023-04-26
 author: zombieJ
+zhihu_column_url: https://zhuanlan.zhihu.com/p/633333904
```
Comment (afc163) [inline docs/blog/why-not-static.en-US.md:5]: zhihu_url 和 yuque_url 吧，简单点。

File: docs/blog/why-not-static.en-US.md [hunk 11]
```diff
@@ -2,6 +2,8 @@
...
+yuque_column_url: https://www.yuque.com/ant-design/ant-design
```
[the remark this answers is not in the document]
Comment (li-jia-nan) [reply docs/blog/why-not-static.en-US.md:6]: 闲夕大佬说他假期同步一下

[the remark this answers is not in the document]
Comment (afc163) [reply docs/blog/why-not-static.en-US.md:6]: 对，所以这里可以先不加链接。

File: docs/blog/why-not-static.en-US.md (hunk 10 above)
Comment (li-jia-nan) [reply docs/blog/why-not-static.en-US.md]: done~

File: docs/blog/why-not-static.en-US.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (li-jia-nan) [reply docs/blog/why-not-static.en-US.md]: done~

File: .dumi/theme/slots/Content/ColumnCard.tsx [hunk 12]
```diff
@@ -0,0 +1,167 @@
...
+  const { card, bigTitle, cardBody, left, title, subTitle, btn } = useStyle();
+  if (!zhihuLink && !yuqueLink) {
+    return null;
+  }
+  return (
+    <Card css={card} bordered={false}>
+      <h3 css={bigTitle}>{locale.bigTitle}</h3>
+      {zhihuLink && (
+        <div css={cardBody}>
+          <div css={left}>
```
Comment (afc163) [inline .dumi/theme/slots/Content/ColumnCard.tsx:112]: 这里可以给个专栏链接

Comment (li-jia-nan) [reply .dumi/theme/slots/Content/ColumnCard.tsx:112]: 没听懂？是点击这个 div 的时候也跳转过去吗？

Comment (afc163) [reply .dumi/theme/slots/Content/ColumnCard.tsx:112]: 图片

加在这里吧，总感觉这个专栏名应该是可点的。

Comment (li-jia-nan) [reply .dumi/theme/slots/Content/ColumnCard.tsx:112]: 我出去吃个饭，晚上回来改

Comment (li-jia-nan) [reply .dumi/theme/slots/Content/ColumnCard.tsx:119]: done~

File: .dumi/theme/slots/Content/ColumnCard.tsx [hunk 13]
```diff
@@ -143,7 +152,9 @@ const ColumnCard: React.FC<Props> = ({ zhihuLink, yuqueLink }) => {
                 <div css={subTitle}>
                   <YuqueOutlined className="logo yuque-logo" />
                   <RightOutlined className="arrowIcon" />
-                  {locale.yuQue}
+                  <Button target="_blank" href={yuqueLink} className="zl-btn" type="link">
```
Comment (afc163) [inline .dumi/theme/slots/Content/ColumnCard.tsx:155]: 链接不对，要写死。和后面的 Button 不一样。

https://www.zhihu.com/column/c_1564262000561106944
https://www.yuque.com/ant-design/ant-design/

Comment (li-jia-nan) [reply .dumi/theme/slots/Content/ColumnCard.tsx]: done~

**Review by afc163 — APPROVED** [review]

**Pull request merged**